### PR TITLE
fix(deps): add filtered npm audit

### PR DESCRIPTION
**Pull request by g-k**
refs:

mozilla/fxa#303
https://bugzilla.mozilla.org/show_bug.cgi?id=1478025

In https://travis-ci.org/mozilla/fxa-content-server/builds/433763993 failing due to advisories:

```console
Unfiltered advisories:
  https://npmjs.com/advisories/310
  https://npmjs.com/advisories/39
  https://npmjs.com/advisories/48
  https://npmjs.com/advisories/532
  https://npmjs.com/advisories/577
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -145,6 +147,7 @@
     "install": "0.12.1",
     "intern": "4.2.2",
     "leadfoot": "1.7.4",
+    "npm": "^6.4.1",
```
Comment (shane-tomlinson) [inline package.json]: @g-k - do we need npm?

File: package.json [hunk 2]
```diff
@@ -122,6 +123,7 @@
     "webrtc-adapter-test": "0.2.5"
   },
   "devDependencies": {
+    "audit-filter": "^0.3.0",
```
Comment (shane-tomlinson) [inline package.json]: Can you remove the "^" prefix?

File: package.json (hunk 1 above)
Comment (g-k) [reply package.json]: We need a new enough version of npm to have the command `npm audit --json`, upgrading the node version would work too.

File: package.json (hunk 2 above)
Comment (g-k) [reply package.json]: :+1:  

/me wonders if there is a way to override what `npm i --save-dev`

File: package.json (hunk 1 above)
Comment (g-k) [reply package.json]: Let me try specifying this in engines https://docs.npmjs.com/files/package.json#engines

**Comment by g-k** [comment]
Ready for another look. Using engines worked, since travis is failing on the audit results.

**Comment by shane-tomlinson** [comment]
> since travis is failing on the audit results.

Looks like we need to update or replace some deps.

**Comment by shane-tomlinson** [comment]
Closing this and using as a basis for a new PR with a couple of exceptions added. Thanks @g-k!

**Comment by shane-tomlinson** [comment]
Replaced by https://github.com/mozilla/fxa-content-server/pull/6676

**Pull request closed without merging**